aws/resource_aws_ecrpublic_repository_policy_test.go

```diff
@@ -0,0 +1,296 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccAWSEcrPublicRepositoryPolicy_basic(t *testing.T) {
+	randString := acctest.RandString(10)
+	resourceName := "aws_ecrpublic_repository_policy.default"
```

Comment:
We prefer Terraform resource names to be `test` wherever possible.

```suggestion
	resourceName := "aws_ecrpublic_repository_policy.test"
```